game-core/src/main/java/games/strategy/engine/chat/Chat.java

```diff
@@ -44,26 +41,26 @@
   private final Collection<BiConsumer<PlayerName, String>> statusUpdateListeners =
       new ArrayList<>();
 
-  public Chat(final Messengers messengers, final String chatName) {
-    this.localPlayerName = messengers.getLocalNode().getPlayerName();
-    chatTransmitter = new JavaSocketChatTransmitter(this, chatName, messengers);
+  public Chat(final ChatTransmitter chatTransmitter) {
+    this.localPlayerName = chatTransmitter.getLocalPlayerName();
+    this.chatTransmitter = chatTransmitter;
+    chatTransmitter.setChatClient(this);
```

Comment:
Ah, another thing I think is weird.
I really feel like there shouldn't be a need to have this kind of circular dependency, but ok 🤔

Reply:
I said in the commit comment the circular dependency is broken, to one extent it is, but really it is not. More fundamentally, Chat as a constructor arg to ChatTransmitter, and now ChatTransmitter as an arg to Chat, that was an unworkable dependency, hence the setter.